agent/structs/config_entry_inline_certificate.go

```diff
@@ -42,8 +43,13 @@ func (e *InlineCertificateConfigEntry) GetEnterpriseMeta() *acl.EnterpriseMeta {
 }
 func (e *InlineCertificateConfigEntry) GetRaftIndex() *RaftIndex { return &e.RaftIndex }
 
+// Envoy will silently reject any keys that are less than 2048 bytes long
+// https://github.com/envoyproxy/envoy/blob/main/source/extensions/transport_sockets/tls/context_impl.cc#L238
+const MinKeyLength = 2048
```

Comment:
given the FIPS work on enterprise, wondering based on the conditional a couple lines below the linked [envoy line](https://github.com/envoyproxy/envoy/blob/83d633d9ff0f0a9ea550ffda2fef20f7f9f3ede1/source/extensions/transport_sockets/tls/context_impl.cc#L244-L249):

```cpp
        if (rsa_key_length != 2048 && rsa_key_length != 3072 && rsa_key_length != 4096) {
          throw EnvoyException(
              fmt::format("Failed to load certificate chain from {}, only RSA certificates with "
                          "2048-bit, 3072-bit or 4096-bit keys are supported in FIPS mode",
                          ctx.cert_chain_file_path_));
        } 
```

We may want to call: https://github.com/hashicorp/consul/blob/b76c4d7bd556a76d3e0c0847e2c91f734530411c/version/fips.go#L5

and do some additional constraint checks and only allow you to create InlineCertificates with those exact bit lengths if built with FIPS... since in the FIPS-compliant envoy build (and due to the movement towards packaging envoy FIPS alongside dataplane we should be able to detect since we require the whole stack including the Consul servers to be FIPS-compliant at that point) any other key length will result in a broken... WDYT?

Reply:
that's a good call out! lemme modify this a bit

Reply:
One other thing to call out is that you'll want to also do this in the k8s side of things on the gateway validation itself, somewhere like this:

https://github.com/hashicorp/consul-k8s/blob/d3f9b670ab8055f0fc8ea4061c2d3c40abeb047f/control-plane/api-gateway/binding/validation.go#L207-L213

Since we don't do any sort of status syncing to Kubernetes secret objects, invalid certificates referenced by a gateway get statuses reflected back only on the gateway. Rather than attempting a sync and having it be silently rejected by Consul, we should just skip the attempt at InlineCertificate creation and set a status on the Gateway object in Kubernetes saying, "the cert you're referencing has an invalid length" to inform the user.

Reply:
yeah I've got some code over in the `ParseCertificates` function currently that I'm gonna push up after this merges so implementations are similar-ish and will have mostly been code reviewed already